nanome_docking/autodock4/calculations.py

```diff
@@ -1,320 +1,175 @@
+        prepare_dpf42_script = os.path.join(os.path.dirname(__file__), 'prepare_dpf42.py')
+        autodock_input_dpf = tempfile.NamedTemporaryFile(delete=False, suffix=".dpf", dir=self.temp_dir)
+        dock_args = [
+            'conda', 'run', '-n', 'adfr-suite',
+            'python', prepare_dpf42_script,
+            '-l', ligands_file_pdbqt.name,
+            '-r', receptor_file_pdbqt.name,
+            '-o', autodock_input_dpf.name
+        ]
+        nanome.util.Logs.debug("Prepare grid and docking parameter files")
+        subprocess.run(dock_args, cwd=self.temp_dir)
+        assert open(autodock_input_dpf.name).read()
+        return autodock_input_dpf
+
+    def _start_autogrid4(self, autogrid_input_gpf):
+        # Start Grid
+        autogrid_log = tempfile.NamedTemporaryFile(delete=False, suffix=".glg", dir=self.temp_dir)
+        args = [
+            'conda', 'run', '-n', 'adfr-suite',
+            'autogrid4', '-p', autogrid_input_gpf.name, '-l', autogrid_log.name
+        ]
+        nanome.util.Logs.debug("Start Autogrid")
+        subprocess.run(args, cwd=self.temp_dir)
+        generated_filepaths = [
+            f'{self.temp_dir}/{filename}' for filename in os.listdir(self.temp_dir)
+            if filename.endswith('.map') or filename.endswith('.fld')
+        ]
+        return generated_filepaths
+
+    def _start_vina(self, receptor_file_pdbqt, ligands_file_pdbqt, num_modes=5, exhaustiveness=8):
+        # Start VINA Docking, using the autodock4 scoring.
+        vina_binary = os.path.join(os.path.dirname(__file__), 'vina_1.2.2_linux_x86_64')
+        dock_results = tempfile.NamedTemporaryFile(delete=False, dir=self.temp_dir, suffix='.pdbqt')
+        # map files created by autogrid call, and are found using the receptor file name.
+        maps_identifier = receptor_file_pdbqt.name.split('.pdbqt')[0]
+        args = [
+            vina_binary,
+            '--scoring', 'ad4',
+            '--maps', maps_identifier,
+            '--ligand', ligands_file_pdbqt.name,
+            '--out', dock_results.name,
+            '--exhaustiveness', str(exhaustiveness),
+            '--num_modes', str(num_modes)
+        ]
+        nanome.util.Logs.debug("Start Autodock")
+        subprocess.run(args, cwd=self.temp_dir)
+        assert open(dock_results.name).read()
+        return dock_results
+
+    def make_complexes_invisible(self, complexes):
+        for comp in complexes:
+            comp.visible = False
+        self._plugin.update_structures_shallow(complexes)
+
+    def convert_pdbqt_to_sdf(self, pdbqt_file):
+        output_file = tempfile.NamedTemporaryFile(delete=False, dir=self.temp_dir, suffix=".sdf")
+        cmd = ['obabel', '-ipdbqt', pdbqt_file.name, f'-O{output_file.name}']
+        subprocess.run(cmd, cwd=self.temp_dir)
+        assert open(output_file.name).read()
+        return output_file
 
-        # shutil.rmtree(self.temp_dir.name)
+    def update(self):
+        pass
```

Comment:
Is this needed?

Reply:
Yep, this is needed so that this line doesn't fail
https://github.com/nanome-ai/plugin-docking/blob/master/nanome_docking/Docking.py#L95